src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -509,77 +393,80 @@ public HttpResponseMessage Annotate(AnnotationPostedDataEntity annotateDocumentR
                         }
                     }
-                    catch (System.Exception ex)
-                    {
-                        throw new System.Exception(ex.Message, ex);
-                    }
                 }
 
                 // Add annotation to the document
-                DocumentType type = DocumentTypesConverter.GetDocumentType(documentType);              
-                RemoveAnnotations(documentGuid);              
+                RemoveAnnotations(documentGuid, password);
                 // check if annotations array contains at least one annotation to add
                 if (annotations.Count != 0)
                 {
-                    Stream cleanDoc = new FileStream(documentGuid, FileMode.Open, FileAccess.ReadWrite, FileShare.ReadWrite);
-                    Stream result = AnnotationImageHandler.ExportAnnotationsToDocument(cleanDoc, annotations, type);
-                    cleanDoc.Dispose();
-                    cleanDoc.Close();
-                    File.Delete(documentGuid);
-                    // Save result stream to file.               
-                    if (annotateDocumentRequest.print)
+                    using (FileStream outputStream = File.OpenRead(documentGuid))
                     {
-                        documentGuid = documentGuid.Replace(System.IO.Path.GetFileNameWithoutExtension(documentGuid), System.IO.Path.GetFileNameWithoutExtension(documentGuid) + "Temp");
+                        using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(outputStream, GetLoadOptions(password)))
+                        {
+                            foreach (var annotation in annotations)
+                            {
+                                annotator.Add(annotation);
+                            }
+
+                            annotator.Save(tempPath);
+                        }
                     }
-                    using (FileStream fileStream = new FileStream(documentGuid, FileMode.Create))
+
+                    if (File.Exists(documentGuid))
                     {
-                        byte[] buffer = new byte[result.Length];
-                        result.Seek(0, SeekOrigin.Begin);
-                        result.Read(buffer, 0, buffer.Length);
-                        fileStream.Write(buffer, 0, buffer.Length);
-                        fileStream.Close();
+                        File.Delete(documentGuid);
                     }
+
+                    File.Move(tempPath, documentGuid);
                 }
 
                 annotatedDocument = new AnnotatedDocumentEntity();
                 annotatedDocument.guid = documentGuid;
                 if (annotateDocumentRequest.print)
                 {
-                    annotatedDocument.pages = GetAnnotatedPagesForPrint(documentGuid);
+                    // TODO: reconsider following method
+                    //annotatedDocument.pages = GetAnnotatedPagesForPrint(documentGuid);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372470)